src/index.js

```diff
@@ -21,114 +19,96 @@ class NanoClamp extends PureComponent {
 
-  clampLines() {
-    this.setState({text: ''}, () => {
-      const maxHeight = this.lineHeight * this.props.lines + 1
-
-      this.start = 0
-      this.middle = 0
-      this.end = this.original.length
-
-      while (this.start <= this.end) {
-        this.middle = Math.floor((this.start + this.end) / 2)
-        this.element.innerText = this.original.slice(0, this.middle)
-        if (this.middle === this.original.length) {
-          this.setState({
-            text: this.original,
-            noClamp: true
-          })
-          return
-        }
-
-        this.moveMarkers(maxHeight)
-      }
+  clampLines () {
+    const maxHeight = this.lineHeight * this.props.lines + 1
 
-      this.setState(
-        () => {
-          const slicedString = this.original.slice(0, this.middle - 5)
-          const text = slicedString.trim() + ELLIPSIS
-          return {
-            text
-          }
-        },
-        () => {
-          this.element.innerText = this.state.text
-        }
-      )
-    })
-  }
+    this.start = 0
+    this.middle = 0
+    this.end = this.original.length
 
-  moveMarkers(maxHeight) {
-    if (this.element.clientHeight <= maxHeight) {
-      this.start = this.middle + 1
-    } else {
-      this.end = this.middle - 1
-    }
-  }
+    while (this.start <= this.end) {
+      this.middle = Math.floor((this.start + this.end) / 2)
+      this.element.innerText = this.original.slice(0, this.middle)
 
-  render() {
-    const {className, is, text: propText} = this.props
-    const {text} = this.state
+      if (this.middle === this.original.length) {
+        this.setState({ text: this.original, noClamp: true })
+        return
+      }
 
-    if (!propText) {
-      return null
+      this.moveMarkers(maxHeight)
     }
 
-    const props = {
-      className,
-      ref: e => {
-        this.element = e
+    this.setState(
+      () => {
+        const slicedString = this.original.slice(0, this.middle - 5)
+        const text = slicedString.trim() + this.props.ellipsis
+        return { text }
+      },
+      () => {
+        this.element.innerText = this.state.text
       }
-    }
+    )
+  }
+
+  moveMarkers (maxHeight) {
+    if (this.element.clientHeight <= maxHeight) this.start = this.middle + 1
+    else this.end = this.middle - 1
+  }
+
+  render () {
+    const {is, text: propText, ...props} = this.props
+    const {text} = this.state
 
-    return createElement(is, props, text)
+    return propText
+      ? createElement(is, { ref: node => (this.element = node), ...props }, text)
+      : null
   }
 }
 
 NanoClamp.defaultProps = {
-  className: '',
   is: 'div',
-  lines: 3
+  lines: 3,
+  ellipsis: '…',
+  debounce: 200
 }
 
 NanoClamp.propTypes = {
-  className: PropTypes.string,
   is: PropTypes.string,
   lines: PropTypes.number,
-  text: PropTypes.string.isRequired
+  debounce: PropTypes.number,
+  text: PropTypes.string.isRequired,
+  ellipsis: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
```

Comment:
What would be the need for `PropTypes.object`?

Reply:
true, not necessary

Reply:
for rendering a react component as ellipsis